src/Workspaces/Core/Portable/CodeStyle/NotificationOption.cs

```diff
@@ -6,33 +6,48 @@
 
 using System;
 using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CodeStyle
 {
     /// <inheritdoc cref="NotificationOption2"/>
-    public class NotificationOption
+    public sealed class NotificationOption
     {
         private readonly NotificationOption2 _notificationOptionImpl;
 
-        /// <inheritdoc cref="NotificationOption2.Name"/>
+        /// <summary>
+        /// Name for the notification option.
+        /// </summary>
         public string Name
         {
-            get => _notificationOptionImpl.Name;
-            set => _notificationOptionImpl.Name = value;
+            get => Severity switch
+            {
+                ReportDiagnostic.Suppress => WorkspacesResources.None,
+                ReportDiagnostic.Hidden => WorkspacesResources.Refactoring_Only,
+                ReportDiagnostic.Info => WorkspacesResources.Suggestion,
+                ReportDiagnostic.Warn => WorkspacesResources.Warning,
+                ReportDiagnostic.Error => WorkspacesResources.Error,
+                _ => throw ExceptionUtilities.UnexpectedValue(Severity)
+            };
+
+            [Obsolete("Modifying a NotificationOption is not supported.", error: true)]
+            set => throw new InvalidOperationException();
```

Comment:
Are there any external users of the obsolete APIs? If they had valid use cases it would be nice to introduce `WithXX` APIs as replacement.

Reply:
I don't see the usage scenario. There are 5 predefined instances of this type and the constructor is private.